tests/test_linter/test_ignore_types.py

```diff
@@ -0,0 +1,472 @@
+
+    messages = [p.message for p in report]
+
+    assert len(messages) == 0
+
+
+# todo PVs are not checked for titles yet
+
+
+def test_class_missing_recommendeds_forbidden():
+    config = yaml.safe_load(
+        """
+rules:
+  recommended:
+    level: error
+"""
+    )
+
+    builder = SchemaBuilder()
+    builder.add_class("MyClass")
+
+    linter = Linter(config)
+    report = list(linter.lint(builder.schema))
+
+    messages = [p.message for p in report]
+
+    assert len(messages) == 1
+    assert any("'MyClass' does not have recommended slot 'description'" in m for m in messages)
+
+
+def test_class_missing_recommendeds_allowed():
+    config = yaml.safe_load(
+        """
+rules:
+  recommended:
+    level: error
+    exclude_type:
+      - class_definition
+"""
+    )
+
+    builder = SchemaBuilder()
+    builder.add_class("MyClass")
+
+    linter = Linter(config)
+    report = list(linter.lint(builder.schema))
+
+    messages = [p.message for p in report]
+
+    assert len(messages) == 0
+
+
+def test_slot_missing_recommendeds_forbidden():
+    config = yaml.safe_load(
+        """
+rules:
+  recommended:
+    level: error
+"""
+    )
+
+    builder = SchemaBuilder()
+    builder.add_slot("my_slot")
+
+    linter = Linter(config)
+    report = list(linter.lint(builder.schema))
+
+    messages = [p.message for p in report]
+
+    print("\n")
+    for m in messages:
+        print(m)
+
+    assert len(messages) == 1
+    assert any("Slot 'my_slot' does not have recommended slot 'description'" in m for m in messages)
+
+
+def test_slot_missing_recommendeds_allowed():
+    config = yaml.safe_load(
+        """
+rules:
+  recommended:
+    level: error
+    exclude_type:
+      - slot_definition
+"""
+    )
+
+    builder = SchemaBuilder()
+    builder.add_slot("my_slot")
+
+    linter = Linter(config)
+    report = list(linter.lint(builder.schema))
+
+    messages = [p.message for p in report]
+
+    assert len(messages) == 0
+
+
+def test_enum_missing_recommendeds_forbidden():
+    config = yaml.safe_load(
+        """
+rules:
+  recommended:
+    level: error
+"""
+    )
+
+    builder = SchemaBuilder()
+    builder.add_enum("MyEnum")
+
+    linter = Linter(config)
+    report = list(linter.lint(builder.schema))
+
+    messages = [p.message for p in report]
+
+    assert len(messages) == 1
+    assert any("Enum 'MyEnum' does not have recommended slot 'description'" in m for m in messages)
+
+
+def test_enum_missing_recommendeds_allowed():
+    config = yaml.safe_load(
+        """
+rules:
+  recommended:
+    level: error
+    exclude_type:
+      - enum_definition
+"""
+    )
+
+    builder = SchemaBuilder()
+    builder.add_enum("MyEnum")
+
+    linter = Linter(config)
+    report = list(linter.lint(builder.schema))
+
+    messages = [p.message for p in report]
+
+    assert len(messages) == 0
+
+
+# todo PVs are not checked for recommended fields yet
+
+
+def test_class_default_non_standard_name_forbidden():
```

Comment:
Recommend:

* Drop `test_*_default_non_standard_name_forbidden` functions (already tested in `test_rule_standard_naming.py`)
* Move `test_*_default_non_standard_name_allowed` functions (consider renaming) into `test_rule_standard_naming.py`